js/components/scripts/compile-translations.js

```diff
@@ -0,0 +1,210 @@
+    console.warn(`⚠️  No .ftl files found in ${localeDir}`);
+    return "";
+  }
+
+  const contents = [];
+
+  for (const filePath of ftlFiles) {
+    try {
+      const content = fs.readFileSync(filePath, "utf-8");
+      const relativePath = path.relative(localeDir, filePath);
+
+      contents.push(`# === ${relativePath} ===`);
+      contents.push(content.trim());
+      contents.push(""); // Empty line separator
+    } catch (error) {
+      console.error(`❌ Error reading ${filePath}:`, error.message);
+    }
+  }
+
+  return contents.join("\n");
+}
+
+/**
+ * Parse FTL content and generate JSON with key-value pairs
+ * Handles multiline constructs like pluralization rules
+ */
+function generateJsonFromFtl(content) {
+  const translations = {};
+  const lines = content.split("\n");
+
+  let currentKey = null;
+  let currentValue = [];
+  let insideMultiline = false;
+  let braceCount = 0;
+
+  for (let i = 0; i < lines.length; i++) {
+    const line = lines[i];
+    const trimmed = line.trim();
+
+    // Skip comments and empty lines when not inside a multiline construct
+    if (
+      !insideMultiline &&
+      (!trimmed || trimmed.startsWith("#") || trimmed.startsWith("##"))
+    ) {
+      continue;
+    }
+
+    // Check for new key = value pair
+    const keyMatch = trimmed.match(/^([a-zA-Z][a-zA-Z0-9_-]*)\s*=\s*(.*)$/);
+
+    if (keyMatch && !insideMultiline) {
+      // Save previous key if exists
+      if (currentKey) {
+        translations[currentKey] = currentValue.join("\n").trim();
+      }
+
+      const [, key, value] = keyMatch;
+      currentKey = key;
+      currentValue = [value];
+
+      // Check if this starts a multiline construct (contains opening brace)
+      const openBraces = (value.match(/\{/g) || []).length;
+      const closeBraces = (value.match(/\}/g) || []).length;
+      braceCount = openBraces - closeBraces;
+
+      if (braceCount > 0) {
+        insideMultiline = true;
+      } else {
+        // Single line value, save immediately
+        translations[currentKey] = value.trim();
+        currentKey = null;
+        currentValue = [];
+      }
+    } else if (insideMultiline && currentKey) {
+      // Continue accumulating multiline value
+      currentValue.push(line); // Preserve original formatting including indentation
+
+      // Count braces to detect end of multiline construct
+      const openBraces = (line.match(/\{/g) || []).length;
+      const closeBraces = (line.match(/\}/g) || []).length;
+      braceCount += openBraces - closeBraces;
+
+      if (braceCount <= 0) {
+        // End of multiline construct
+        translations[currentKey] = currentValue.join("\n").trim();
+        currentKey = null;
+        currentValue = [];
+        insideMultiline = false;
+        braceCount = 0;
+      }
+    }
+  }
+
+  // Handle any remaining key
+  if (currentKey) {
+    translations[currentKey] = currentValue.join("\n").trim();
+  }
+
+  return JSON.stringify(translations, null, 2);
+}
```

Comment:
Surely there's a library for this so we don't have to hand-roll it? Looking at https://github.com/projectfluent/fluent.js